core/src/main/java/com/graphhopper/PathWrapper.java

```diff
@@ -243,6 +246,23 @@ public void setInstructions(InstructionList instructions) {
         this.instructions = instructions;
     }
 
+    public void addPathDetails(List<PathDetails> details){
+        if(this.pathDetails.isEmpty()){
+            this.pathDetails = details;
+        }else{
+            if(this.pathDetails.size() != details.size()){
+                throw new IllegalStateException("Details have to be the same size");
+            }
+            for (int i = 0; i < details.size(); i++) {
+                this.pathDetails.get(i).merge(details.get(i));
+            }
+        }
+    }
```

Comment:
See below: what is a 'merge'? And why not just adding PathDetails like the method says it?

Also instead of EMPTY_LIST and if+else I would prefer without the if and always do the same.